x-pack/plugins/infra/public/components/logging/log_entry_flyout/log_entry_actions_menu.tsx

```diff
@@ -7,38 +7,39 @@
 import { EuiButtonEmpty, EuiContextMenuItem, EuiContextMenuPanel, EuiPopover } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n/react';
 import React, { useMemo } from 'react';
-import url from 'url';
-import { useKibana } from '../../../../../../../src/plugins/kibana_react/public';
 import { useVisibilityState } from '../../../utils/use_visibility_state';
 import { getTraceUrl } from '../../../../../../legacy/plugins/apm/public/components/shared/Links/apm/ExternalLinks';
 import { LogEntriesItem } from '../../../../common/http_api';
+import { useLinkProps, LinkDescriptor } from '../../../hooks/use_link_props';
 
 const UPTIME_FIELDS = ['container.id', 'host.ip', 'kubernetes.pod.uid'];
 
 export const LogEntryActionsMenu: React.FunctionComponent<{
   logItem: LogEntriesItem;
 }> = ({ logItem }) => {
-  const prependBasePath = useKibana().services.http?.basePath?.prepend;
   const { hide, isVisible, show } = useVisibilityState(false);
 
-  const uptimeLink = useMemo(() => {
-    const link = getUptimeLink(logItem);
-    return prependBasePath && link ? prependBasePath(link) : link;
-  }, [logItem, prependBasePath]);
+  const apmLinkDescriptor = useMemo(() => getAPMLink(logItem), [logItem]);
+  const uptimeLinkDescriptor = useMemo(() => getUptimeLink(logItem), [logItem]);
 
-  const apmLink = useMemo(() => {
-    const link = getAPMLink(logItem);
-    return prependBasePath && link ? prependBasePath(link) : link;
-  }, [logItem, prependBasePath]);
+  const uptimeLinkProps = useLinkProps({
+    app: 'uptime',
```

Comment:
I've moved the `app` key into the separate functions now, but I've also left them here, this is because of the conditional nature of the code below this, and the fact hooks can't be called within a conditional. The `disabled` prop ensures the items are disabled based on the descriptor result. This isn't 100% ideal, but I can't think of a way around it without introducing complexity that (imo) just isn't warranted here for links.